website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,214 @@
+
+- [Test warnings](/reference/resource-configs/severity) with the `--warn-error-options '{"error": ["LogTestResults"]}'` flag.
+- Jinja [exception warnings](/reference/dbt-jinja-functions/exceptions#warn) with `--warn-error-options '{"error": ["JinjaLogWarning"]}'`.
+- No nodes selected with `--warn-error-options '{"error": ["NoNodesForSelectionCriteria"]}'`.
+- Adapter deprecation warnings with `--warn-error-options '{"error": ["AdapterDeprecationWarning"]}'`.
 
-Consider using [`--warn-error-options`](#using---warn-error-options-for-targeted-warnings) for more targeted control over which warnings are treated as errors. As an example, if you'd like to only promote test warnings to errors, set that flag to `--warn-error-options '{"error": ["LogTestResult"]}'`.
+</VersionBlock>
+
+<VersionBlock lastVersion="1.7">
+
+:::caution Proceed with caution in production environments
+Using the `--warn-error` flag or `--warn-error-options '{"include": "all"}'` will treat _all_ current and future warnings as errors.
+
+This means that if a new warning is introduced in a future version of dbt Core, your production job may start failing unexpectedly. We recommend proceeding with caution when doing this in production environments, and explicitly listing only the warnings you want to treat as errors in production.
+:::
+
+Instead of using the `--warn-error` flag to promote _all_ warnings to errors, you can use [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag to promote _specific_ warnings to errors, including:
+
+- Test warnings with the `--warn-error-options '{"include": ["LogTestResults"]}'` flag.
+- Jinja-level warnings with the `--warn-error-options '{"include": ["JinjaLogWarning"]}'` flag or [`exceptions.warn`](/reference/dbt-jinja-functions/exceptions#warn).
+- Selection issues with the `--warn-error-options '{"include": ["NoNodesForSelectionCriteria"]}'` flag.
+- Adapter deprecation warnings with the `--warn-error-options include": ["AdapterDeprecationWarning"]}'` flag.
+
+</VersionBlock>
 
 <File name='Usage'>
 
-```text
-dbt --warn-error run
-...
-```
+  ```text
+  dbt --warn-error run
+  ```
 
 </File>
 
-### Using `--warn-error-options` for targeted warnings
+## Use `--warn-error-options` for targeted warnings
 
-Converting any warnings to errors may suit your needs perfectly, but there may be some warnings you just don't care about, and some you care about a lot. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
+In some cases, you may want to convert _all_ warnings to errors. However, when you want _some_ warnings to stay as warnings and only promote or silence specific warnings you can instead use `--warn-error-options`. The `WARN_ERROR_OPTIONS` config or `--warn-error-options` flag gives you more granular control over _exactly which types of warnings_ are treated as errors. 
```

Comment:
i don't know why we capitalize WARN_ERROR_OPTIONS config when the actual parameter is `warn_error_options`. maybe it's just me but capitalizing it makes it seem like it's different?

Reply:
I was trying to make the distinction between environment variable configs (all caps) and flags (`--lowercase`) from what I read in [About flags](https://docs.getdbt.com/reference/global-configs/about-global-configs#setting-flags) page and noticed in other flag pages.

BUT do this inconsistently in the other available flags: [logs](https://docs.getdbt.com/reference/global-configs/logs) vs [indirect selection](https://docs-getdbt-com-git-update-warn-error-dbt-labs.vercel.app/reference/global-configs/indirect-selection). Maybe we stick with lowercase and link to the Env variables page for more info?

Reply:
hmm yeah the flag is lowercase (`--warn-error-options`), but the env var is all caps (`WARN_ERROR_OPTIONS`)